shared/chat/conversation/input/mention-handlers.hoc.js

```diff
@@ -0,0 +1,243 @@
+// @flow
+import * as React from 'react'
+import {type Props} from '.'
+import logger from '../../../logger'
+
+type MentionHocProps = {
+  channelMentionFilter: string,
+  channelMentionPopupOpen: boolean,
+  setChannelMentionFilter: (filter: string) => void,
+  setChannelMentionPopupOpen: (setOpen: boolean) => void,
+  mentionFilter: string,
+  mentionPopupOpen: boolean,
+  setMentionFilter: (filter: string) => void,
+  setMentionPopupOpen: (setOpen: boolean) => void,
+  _inputSetRef: any => void,
+} & Props
+
+type MentionHocState = {
+  upArrowCounter: number,
+  downArrowCounter: number,
+  pickSelectedCounter: number,
+}
+
+const mentionHoc = (InputComponent: React.ComponentType<Props>) => {
```

Comment:
I was unsure of the best way to type this, I went with what we do in `header-hoc`